pkg/cli/rpc_client.go

```diff
@@ -45,7 +44,7 @@ func getClientGRPCConn(
 	stopper := stop.NewStopper(stop.WithTracer(tracer))
 	rpcContext := rpc.NewContext(ctx,
 		rpc.ContextOptions{
-			TenantID: roachpb.SystemTenantID,
+			TenantID: cfg.TenantID,
```

Comment:
I believe this is the change that makes everything break. 

You will want to continue to use `roachpb.SystemTenantID` here. 
I'd admit this is a tad confusing. Generally, CLI tools do not identify "as a tenant"—neither as the system tenant or a secondary tenant. They just identify as a _RPC client_ which does not have any tenant identity.

However, the rpc.Context uses a shared code path for both the case where a client connects (your code, here) and the case where a KV node connects to another KV node. That is the code path you want to exercise here.

Reply:
I am not sure I understand. For the KV node, would this not resolve to the system tenant ID. How else does a CLI tool make an rpc request to a tenant specifically? My understanding was that the rpc context needs to reflect the tenant ID of the desired target, is that incorrect?

Reply:
> How else does a CLI tool make an rpc request to a tenant specifically?

_Currently_ (until #92524 is resolved) there is a separate RPC port per tenant. You make a RPC request to a tenant specifically by connecting to its RPC port. That is how the tenant is selected. When you do this, you still use roachpb.SystemTenantID in the ContextOptions here.

When #92524 is resolved, there will be a _separate mechanism_ to select the target tenant, probably something called `TargetTenantID` in the context options. We'll change your code when we fix #92524.

Reply:
Ok cool, thanks. Sorry I lost track of where we were with the unification of ports. It is the SQL ports that are shared across tenants currently not the RPC. Thanks for the additional context!